File: src/main/java/dao/UsuarioDao.java
package dao;

import kontroller.Usuario;
import java.io.Serializable;

public class UsuarioDao extends AbstractDao<Usuario> implements Serializable {
	private static final long serialVersionUID = -118468822692102503L;

	public Usuario findUserWith(String username, String password) {
		return getEntityManager().createQuery("select u from Usuario u" +
				" where u.username = :username" +
				" and   u.password = :password", Usuario.class)
				.setParameter("username", username)
				.setParameter("password", password)
				.getResultList()
				.stream()
				.findFirst()
				.orElse(null);
	}
}

File: src/main/java/dao/UsuarioDao.kt
package dao

 import model.Usuario
 import java.io.Serializable
 import javax.persistence.NoResultException
 import javax.persistence.TypedQuery

open class UsuarioDao: AbstractDao<Usuario>(Usuario::class.java),Serializable {
    
    fun findUserWith(username: String, password: String): Usuario {
        return getEntityManager().createQuery("select u from Usuario u" +
                " where u.username = :username" +
                " and   u.password = :password", Usuario::class.java)
                .setParameter("username", username)
                .setParameter("password", password)
                .safeSingleResult()
    }

   fun TypedQuery<Usuario>.safeSingleResult(): model.Usuario {
        return try {
            this.singleResult
        } catch (e: NoResultException) {
            e.printStackTrace()
            Usuario()
        }
    }}